universal-application-tool-0.0.1/app/controllers/admin/QuestionController.java

```diff
@@ -117,42 +74,82 @@ public Result newOne(Request request, String type) {
   }
 
   @Secure(authorizers = Authorizers.Labels.UAT_ADMIN)
-  public CompletionStage<Result> index(Request request) {
-    Optional<String> maybeFlash = request.flash().get("message");
+  public Result create(Request request, String questionType) {
+    QuestionForm questionForm;
+    try {
+      questionForm = createQuestionForm(request, questionType);
+    } catch (InvalidQuestionTypeException e) {
+      // Invalid question type.
+      return badRequest(e.toString());
+    }
+
+    QuestionDefinition questionDefinition;
+    try {
+      questionDefinition = questionForm.getBuilder().setVersion(1L).build();
+    } catch (UnsupportedQuestionTypeException e) {
+      // Valid question type that is not yet fully supported.
+      return badRequest(e.toString());
+    }
+
+    ErrorAnd<QuestionDefinition, CiviFormError> result = service.create(questionDefinition);
+    if (result.isError()) {
+      String errorMessage = joinErrors(result.getErrors());
+      return ok(editView.renderNewQuestionForm(request, questionForm, errorMessage));
```

Comment:
question: does this render the same form the admin just filled out, with the data they entered so they can fix it?

Reply:
Exactly